stable/mysqldump/templates/configmap.yaml

```diff
@@ -0,0 +1,107 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "mysqldump.fullname" . }}-script
+  labels:
+    app: {{ template "mysqldump.name" . }}
+    chart: {{ template "mysqldump.chart" . }}
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+data:
+  backup.sh: |-
+    #!/bin/sh
+    #
+    # mysql backup script
+    #
+    {{ if .Values.debug }}
+    set -ex
+    {{ end }}
+    BACKUP_DIR="/backup"
+    TIMESTAMP="$(date +%Y%m%d%H%M%S)"
+
+    echo "test mysql connection"
+    if [ -z "$(mysql -h ${MYSQL_HOST} -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} -B -N -e 'SHOW DATABASES;')" ]; then
+      echo "mysql connection failed! exiting..."
+      exit 1
+    fi
+
+    echo "started" > ${BACKUP_DIR}/${TIMESTAMP}.state
+
+    {{- if (.Values.persistence.enabled) or (.Values.persistentVolumeClaim) }}
+
+    {{ if .Values.housekeeping.enabled }}
+    echo "delete old backups"
+    find ${BACKUP_DIR} -maxdepth 1 -mtime +${KEEP_DAYS} -regex "^${BACKUP_DIR}/[0-9]*_.*\.sql\.gz$" -type f -exec rm {} \;
+    {{ end -}}
+
+    {{ if and (.Values.mysql.db) (eq .Values.allDatabases.enabled false) }}
+    MYSQL_DB="{{ .Values.mysql.db }}"
+    echo "Backing up single db ${MYSQL_DB}"
+    mysqldump ${MYSQL_OPTS} -h ${MYSQL_HOST} -P ${MYSQL_PORT} -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} ${MYSQL_DB} | gzip > ${BACKUP_DIR}/${TIMESTAMP}_${MYSQL_DB}.sql.gz
+
+    {{ else if and (.Values.allDatabases.enabled) (eq .Values.allDatabases.SingleSqlFile false)}}
+    for MYSQL_DB in $(mysql -h "${MYSQL_HOST}" -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} -B -N -e "SHOW DATABASES;"|egrep -v '^(information|performance)_schema$'); do
+      echo "Backing up db ${MYSQL_DB}"
+      mysqldump ${MYSQL_OPTS} -h ${MYSQL_HOST} -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} ${MYSQL_DB} | gzip > ${BACKUP_DIR}/${TIMESTAMP}_${MYSQL_DB}.sql.gz
+      rc=$?
+    done
+
+    {{ else if and (.Values.allDatabases.enabled) (.Values.allDatabases.SingleSqlFile) }}
+    echo "Backing up all databases"
+    MYSQL_DB="alldatabases"
+    mysqldump ${MYSQL_OPTS} -h ${MYSQL_HOST} -P ${MYSQL_PORT} -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} --all-databases | gzip > ${BACKUP_DIR}/${TIMESTAMP}_${MYSQL_DB}.sql.gz
+    rc=$?
+    {{- end -}}
+
+    {{- if (.Values.upload.googlestoragebucket.enabled) or (.Values.upload.ssh.enabled) -}}
+    {{ if .Values.upload.ssh.enabled -}}
+    echo "upload files via ssh"
+    rsync -av --delete --exclude=*.state -e 'ssh -o StrictHostKeyChecking=accept-new -o UserKnownHostsFile=/dev/null' ${BACKUP_DIR}/ {{ .Values.upload.ssh.user }}@{{ .Values.upload.ssh.host }}:/{{ .Values.upload.ssh.dir }}
+    rcu=$?
+    {{ end -}}
+
+    {{ if .Values.upload.googlestoragebucket.enabled }}
+    echo "upload files to google storage bucket"
```

Comment:
Would be good to have the destination logged.

Reply:
done